src/nci/NfcManager.cpp

```diff
@@ -695,8 +712,16 @@ static void nfaConnectionCallback(UINT8 connEvent, tNFA_CONN_EVT_DATA* eventData
     // NFC link/protocol activated.
     case NFA_ACTIVATED_EVT:
       ALOGD("%s: NFA_ACTIVATED_EVT: gIsSelectingRfInterface=%d, sIsDisabling=%d", __FUNCTION__, gIsSelectingRfInterface, sIsDisabling);
-      if (sIsDisabling || !sIsNfaEnabled)
+#ifdef NFCC_PN547
+      if (EXTNS_GetConnectFlag()) {
+        NfcTag::getInstance().setActivationState();
+        NfcTagManager::doConnectStatus(true);
+        break;
+      }
+#endif
```

Comment:
The source code provided by NXP calling this before  |if (sIsDisabling || ... )| , I think EXTNS_GetConnectFlag won't be true if NFC is disabled.
I will move this after  |if (sIsDisabling || ... )| for better logic